packages/slider/src/ux-slider.ts

```diff
@@ -98,36 +98,52 @@ export class UxSlider implements UxComponent {
   }
 
   public updateValue(currentMouseX: number) {
-    const normalizedMouseX = currentMouseX - this.element.offsetLeft;
-    const percentValue = normalizedMouseX / this.element.clientWidth;
+    const rect = this.element.getBoundingClientRect();
+    const normalizedMouseX = currentMouseX - rect.x;
+    const percentValue = normalizedMouseX / rect.width;
     const rawValue = ((this.max - this.min) * percentValue) + this.min;
     const numSteps = Math.round((rawValue - this.min) / this.step);
     const steppedValue = this.min + (this.step * numSteps);
 
     this.value = this.boundValue(steppedValue);
   }
 
-  public onTrackMouseDown() {
+  public onTrackMouseDown(e: MouseEvent | TouchEvent) {
     if (this.disabled) {
       return;
     }
 
     this.isActive = true;
 
     const winEvents = new ElementEvents(window as any);
-    const upAction = (e: MouseEvent) => {
+    const upAction = (e: MouseEvent | TouchEvent | FocusEvent) => {
       if (!this.isActive) {
+        winEvents.disposeAll();
         return;
       }
-
-      this.updateValue(e.clientX);
+      if (e instanceof MouseEvent && e.clientX) {
```

Comment:
let's extract these `if`s into a variable:
```ts
const isMouseEvent = e instanceof MouseEvent && e.clientX > 0;
```
is the `e.clientX > 0` needed?

Reply:
We also listen to `FocusEvent` triggered by `blur`, which also lead to an `upAction` but without updating the value because we don't have a value to use in this type of Event.

I'm not sure that the `e.clientX > 0` is needed. I will run some tests to make sure, but I believe we can remove it.

I'll extract te `ifs into variables 👍

Reply:
In fact if I extract the `if`s into variables I get Typescript errors inside the `if`s. 

`e.clientX` only belongs to MouseEvent
and
`e.touches[0].clientX` only belongs to TouchEvent

Maybe better to leave it as it is to hint typescript what will be inside the `if`

Reply:
@bigopon 
Fix: https://github.com/platform5/ux/commit/f41e9dd463d29a0a0a66ead4e10b57fe50b31183

Reply:
For the TS error, we can do a cast in side if blocks:
````ts
if (isMouseEvent) {
  (e as MouseEvent).clientX ...
} else {
  (e as TouchEvent).touches[0].clientX...
}
```

Reply:
@bigopon https://github.com/aurelia/ux/pull/232/commits/6af03b54495ae307905c35b69248d9e6c91f2086